File: src/main/java/com/project/Car_Rental_System/services/owner/OwnerService.java
package com.project.Car_Rental_System.services.owner;

import com.project.Car_Rental_System.dto.BookACarDto;
import com.project.Car_Rental_System.dto.CarDto;
import com.project.Car_Rental_System.dto.CarDtoListDto;
import com.project.Car_Rental_System.dto.SearchCarDto;
import com.project.Car_Rental_System.entity.Car;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface OwnerService {

    boolean postCar(CarDto carDto);

    List<CarDto> getAllCars();

    void deleteCar(Long id);

    CarDto getCarById(Long id);

    List<BookACarDto> getBookings();

    boolean changeBookingStatus(Long bookingId,String status);

    CarDtoListDto searchCar(SearchCarDto searchCarDto);
}

File: src/app/modules/owner/services/owner.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { StorageService } from '../../../auth/services/storage/storage.service';

const BASIC_URL = "http://localhost:8080";

@Injectable({
  providedIn: 'root'
})
export class OwnerService {
  constructor(private http: HttpClient, private storageService: StorageService ) {}

  postCar(carDto: any): Observable<any> {
    return this.http.post(BASIC_URL + "/api/owner/car", carDto, {
        headers: this.createAuthorizationHeader()
    });
  }

  getAllCars(): Observable<any>{
      return this.http.get(BASIC_URL + "/api/owner/cars" ,{
      headers: this.createAuthorizationHeader()
    })
  }

  deleteCar(id:number):Observable<any>{
    return this.http.delete(BASIC_URL + "/api/owner/car/" + id,{
      headers: this.createAuthorizationHeader()
    })
  }

  getCarById(id: number): Observable<any>{
    return this.http.get(BASIC_URL + "/api/owner/car/" + id,{
      headers: this.createAuthorizationHeader()
  });
}
getCarBookings(): Observable<any>{
  return this.http.get(BASIC_URL + "/api/owner/car/booking" ,{
  headers: this.createAuthorizationHeader()
})
}
changeBookingStatus(bookingId: number, status: string): Observable<any> {
  return this.http.get(`${BASIC_URL}/api/owner/car/booking/${bookingId}/${status}`, {
    headers: this.createAuthorizationHeader()
  });
}
searchCar(searchCarDto: any): Observable<any> {
  return this.http.post(BASIC_URL + "/api/owner/car/search", searchCarDto, {
      headers: this.createAuthorizationHeader()
  });
}


  private createAuthorizationHeader(): HttpHeaders {
    const token = StorageService.getToken();  
    console.log("Token Sent:", token); 
  
    let headers = new HttpHeaders();
    if (token) {
      headers = headers.set('Authorization', `Bearer ${token}`);
    } else {
      console.error("No token found!");
    }
    return headers;
  }
}
